src/api/db/data/20250131094818_nullify_targets_on_bs_request_actions.rb

```diff
@@ -3,30 +3,12 @@
 class NullifyTargetsOnBsRequestActions < ActiveRecord::Migration[7.0]
   # rubocop:disable Rails/SkipsModelValidations
   def up
-    bs_request_actions = BsRequestAction.where('target_project_id IS NOT NULL AND target_package_id IS NULL')
-    bs_request_actions.in_batches do |batch|
-      batch.find_each do |action|
-        target_project = Project.find_by(name: action.target_project)
-        if target_project.nil?
-          action.update_columns(target_project_id: nil)
-        end
-      end
+    BsRequestAction.joins('LEFT JOIN projects ON bs_request_actions.target_project_id = projects.id WHERE projects.id IS NULL').find_each do |action|
```

Comment:
if you use `in_batches` you can send a single write to the whole batch with https://api.rubyonrails.org/classes/ActiveRecord/Relation.html#method-i-update_all